File: SimpleCMS/Models/Content.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;
using SimpleCMS.Validations;

namespace SimpleCMS.Models
{
    [Table("T_Content")]
    public class Content
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(255)]
        [Column(TypeName = "nvarchar")]
        public string Title { get; set; }

        [Required]
        public int CategoryId { get; set; }

        [ForeignKey("CategoryId")]
        public virtual Category Category { get; set; }

        [MaxLength(255)]
        [Column(TypeName = "nvarchar")]
        public string Image { get; set; }

        [MaxLength(500)]
        [Column(TypeName = "nvarchar")]
        public string Summary { get; set; }

        [Required]
        [Column(TypeName = "text")]
        public string Body { get; set; }

        [Required]
        [DefaultDateTimeValue("Now")]
        public DateTime? Created { get; set; }

        [Required]
        [DefaultValue(0)]
        public int Hits { get; set; }

        [Required]
        [Range(0, 1)]
        public byte State { get; set; }

        [Required]
        [DefaultValue(0)]
        public int SortOrder { get; set; }

        public virtual ICollection<Tag> Tags { get; set; }

    }
}

File: SimpleCMS/Sencha/app/model/Content.js
Ext.define('SimpleCMS.model.Content', {
    extend: 'SimpleCMS.model.Base',

    fields: [
        { name: 'CategoryId', type: 'auto', defaultValue: null },
        { name: 'CategoryTitle', type: 'string', defaultValue: '' },
        { name: 'Title', type: 'string', defaultValue: '' },
        { name: 'Image', type: 'string', defaultValue: '' },
        { name: 'Summary', type: 'string', defaultValue: '' },
        { name: 'Body', type: 'string', defaultValue: '' },
        { name: 'Tags', type: 'string', defaultValue: '' },
        { name: 'Created', type: 'date', dateFormat: I18N.DefaultDatetimeFormat},
        { name: 'Hits', type: 'int' },
        { name: 'SortOrder', type: 'int', defaultValue: 0 }
    ]
});
